var/spack/repos/builtin/packages/py-pylint/package.py

```diff
@@ -38,20 +40,23 @@ class PyPylint(PythonPackage):
     depends_on('py-astroid@2.2.0:2', type=('build', 'run'), when='@2.3.0:2.7')
     depends_on('py-astroid@2.5.6:2.6', type=('build', 'run'), when='@2.8.0:2.10')
     depends_on('py-astroid@2.8.0:2.8', type=('build', 'run'), when='@2.11.1:')
-
+    depends_on('py-astroid@2.11.2:2.12.0-dev', type=('build', 'run'), when='@2.13.5:')
```

Comment:
```suggestion
    depends_on('py-astroid@2.11.2:2.11', type=('build', 'run'), when='@2.13.5:')
```
Spack's version sorting works differently than pip's, so 2.12.0-dev is actually greater than 2.12.0 in Spack.